frontend/src/main/java/starter/project/frontend/CarController.java

```diff
@@ -8,26 +8,27 @@
 
 @RestController
 public class CarController {
-
-    private List<Car> currentCarList = new ArrayList<>();
     Map<Integer, Car> map = new HashMap<>();
 
     @PostMapping(path = "/car", consumes = {"application/json"})
-    public List<Car> addCar(@RequestBody Car car) {
-        currentCarList.add(car);
-        return currentCarList;
+    public Map<Integer,Car> addCar(@RequestBody Car car) {
+        map.put(map.size()+1,car);
+        car.setId(map.size());
+        return map;
     }
+
     @GetMapping(path = "/car/{carId}")
     public Car getCar(@PathVariable("carId") int carId) {
-        for (Car car : currentCarList) {
+        for (Map.Entry<Integer, Car>  entry : map.entrySet()) {
```

Comment:
Lets avoid iterating through the whole map. If our map is 1000 cars long and our request is only 1 car, then we should only need to check the 1 car. We are not utilising the key feature of maps.

https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#getOrDefault-java.lang.Object-V-

Reply:
thanks